### PR TITLE
Equipment System

### PR DESCRIPTION
**Pull request by GarOfMeridian**
Here's where I'll build the proposed equipment system.

---
**New Armor System**

A new setting can switch over all defense modifiers to a new system in
which Armor is always zero, and defense gear instead adds different
combinations of weapon and magic resistance.

Resistances will be handled differently in this new system - weapon and
magic resistance have become primary resistances that always affect
relevant attack types, while specific elements will then also resist the
remaining damage.

For example, 20% magic resistance and 20% fire resistance will combine
to the equivalent of 36% total resistance against a blast of fire. They're not
adding together - they're both reducing percentages of the damage.
This will let us get creative with spells and skills that reduce enemy
resistances, because we can lower specific elements without borking
the whole system. There is also negative tension between primary
and secondary resists, meaning that you'll need a TON of secondary
resist for it to matter if you've already got high primary resist. You can't,
for example, get a ton of magic resistance and then expect your
fire resistance to really do much unless you get a LOT of it - because
it's acting on a much smaller portion of damage. Also, you really,
really, really don't want negative primary and secondary resists. Really.
They'll both amplify, because you've made some terrible life decisions.
Ex: -20% spell resist and -20% fire resist = -44% resistance to bof.

Partially magical attacks, such as touch spells, are reduced by both
weapon and magic resistance, but at half their values (for example,
touch of flame against someone with 20% weapon resistance and -10% magic
resistance will experience only 5% total primary resistance).

The max strength and min weakness system still applies to secondary resistances.
If you have 20% fire resistance, 10% cold resistance, and -30% shock
resistance, a shock&fire&cold spell will encounter a resistance sum of
-10%. (20 - 30) (Not that any spell like that currently exists, but just to
illustrate).

This is also made possible because percentage resists are now far
more accurate, thanks to Morbus'  upgrade to our health system
(health is counted in thousandths on the server - you can deal
fractions of a hit point in damage).

Both primary and secondary resists are applied 'at the same time', so
there are no unexpected snafus with negative resists increasing or
decreasing damage strangely due to order of damage steps.

Under this new system, Gort and Resist Magic have been lowered to a
maximum of 10% weapon and magic resistance respectively. (With this
system active, gort no longer does anything weird with damage caps or
armor). If this system is not active, Resist Magic is still 5-25% magic
resistance, and Gort still functions as normal. But don't underestimate
the 10% values... they have increasing returns as you get higher in
primary resistance. 10% weapon resist reduces remaining damage by 10% at zero base
resistance, but, thanks to math, 10% weapon resist reduces remaining damage
by *50%* at 80% base resistance (20% remaining damage reduced by 10% is half as much damage). Not that you can really get that high, but it's a striking example.

Values:
Leather: 10% weapon resistance, 30% spell resistance
Chain: 20% weapon resistance, 20% spell resistance
Scale: 30% weapon resistance, 10% spell resistance
Plate: 50% weapon resistance, -10% spell resistance
Nerudite: -10% weapon resistance, 50% spell resistance
DRobes: -10% weapon resistance, -10% spell resistance

All other properties of shields were retained (defense, sp, etc)
Heavy Shields: 10% weapon resistance
Medium Shields: 5% weapon resistance
Light Shields: 0% weapon resistance

Helmet: 5% weapon resistance
Magic Spirit Helm: 5% weapon resistance, 15% spell resistance

Full power Jonas shield: up to 5% weapon resistance

Gort: Up to 10% weapon resistance
RM: Up to 10% spell resistance

General maximum ends of the spectrum:
Plate + Heavy Shield + Helmet + Gort + RM + FPJS = 80% weapon resistance, 0% spell resistance
Neru + Heavy Shield + Helmet + Gort + RM + FPJS = 20% weapon resistance, 60% spell resistance

Partially magical attacks would of course see 40% total resistance in both scenarios. This includes shal/qor enchanted weapons and touch spells.

General maximum ends of the spectrum when Purged & no FPJS:
Plate + Heavy Shield + Helmet = 65% weapon resistance, -10% spell resistance
Neru + Heavy Shield + Helmet = 5% weapon resistance, 50% spell resistance

I think these new values will allow us to create more interesting monsters. Right now, a monster that does fire damage is actually WORSE than a regular monster. It gets no bonus damage, and it is lowered by fire resistance. In this new system, when you're geared up with a ton of weapon resistance, that sudden fire damage (as a partially magical attack) will see half as much resistance as a normal attack - and a blast of fire spell will straight wreck you.

**TL;DR Improved armor and shield offerings while maintaining their inherent 'themes' (Plate good against weapons, Nerudite good against spells, etc). System is also better for content growth, & mobs can be made significantly more varied and dangerous. Additionally, we can potentially get rid of one stat bar (Armor).**

---
**Item Improvers**

Gems with a variety of equipment-improving abilities have been made possible.

* Max Health / Mana / Vigor
* Primary stats (Might, Int, etc)
* Resistances to specific elements
* Damage with specific elements - gear can also contribute to this, such as a new glove item that spawns with +1-10% bonus to a random element
* Blind/Dazzle/Hold dampening - You are more resistant to blind&dazzle or hold, but your use of these spells are also reduced. Ex At 100% visual dampening you are immune to blind, and cannot cast blind or dazzle. This includes any source of these effects - procs, spore, etc.

A system setting enables drops from mobs (5% chance for any given drop to be a gem instead). Drops are dependent on monster level and gem rarity.

A system setting enables sockets on gear dropping from mobs. Monster level and difficulty play into this roll as well.

The new glove item doesn't currently drop anywhere.

I wouldn't activate these systems until more balancing has been done. This is really just a framework to use for determining better values (for both gems and their rarities).

### CODE REVIEW

**Comment by M59lost** [comment]
Too much

**Comment by Daenks** [comment]
@M59lost same thing here. Why is it too much? what do you like? why? what do you dislike why? have any ideas for a better solution? why are your ideas better?

**Comment by M59lost** [comment]
The current armor system should not be replaced by %. 100% or any additional reduction of hold and blind make pure fighters too strong when Anti magic aura is still one of the best spells in the game. All in all gems could work but I dislike the idea of people with more than 150 Hp. I like the current system.

**Comment by treymd** [comment]
Gems are cool, but I'd also like to see 100s (or more) of potential combinations based on stats that are attached to gear itself, before gems are even added.  As far as the "fine the way it is" opinion goes, we need to adapt to and adopt from modern games.  Meridian died years ago from keeping things the way they are.  Slap a defibrillator on it already.
